modules/imgproc/src/filter.cpp

```diff
@@ -4579,7 +4579,7 @@ static bool ipp_filter2D( InputArray _src, OutputArray _dst, int ddepth,
     int stype = src.type(), sdepth = CV_MAT_DEPTH(stype), cn = CV_MAT_CN(stype),
         ktype = kernel.type(), kdepth = CV_MAT_DEPTH(ktype);
     bool isolated = (borderType & BORDER_ISOLATED) != 0;
-    Point ippAnchor(kernel.cols >> 1, kernel.rows >> 1);
+    Point ippAnchor((kernel.cols-1)/2, (kernel.rows-1)/2);
```

Comment:
Looks like this code breaks IPP 8.1.0 support (changed ippAnchor and removed "flip" call):
Failed test: `Imgproc_Filter2D.accuracy` and others

Possible solution is to restrict this optimization for IPP >= 9.0.0 only